src/promptflow-evals/promptflow/evals/evaluators/_eci/_eci.py

```diff
@@ -0,0 +1,88 @@
+    def __init__(self, project_scope: dict, credential=None):
+        self._project_scope = project_scope
+        self._credential = credential
+
+    async def __call__(self, *, question: str, answer: str, **kwargs):
+        # Validate inputs
+        # Raises value error if failed, so execution alone signifies success.
+        if not (question and question.strip() and question != "None") or not (
+            answer and answer.strip() and answer != "None"
+        ):
+            raise ValueError("Both 'question' and 'answer' must be non-empty strings.")
+
+        # Run score computation based on supplied metric.
+        result = await evaluate_with_rai_service(
+            metric_name=_InternalEvaluationMetrics.ECI,
+            question=question,
+            answer=answer,
+            project_scope=self._project_scope,
+            credential=self._credential,
+        )
+        return result
+
+
+class ECIEvaluator:
+    """
+    Initialize an ECI evaluator to evaluate ECI based on the following guidelines:
+    Detects whether ECI is present without a disclaimer in the AI system’s response.
+    After the response is annotated by GPT using the binary annotation prompt
+    (which checks for disclaimer or authoritative sources),
+    we now want to check, if there was a disclaimer, the language should be exactly
+    "AI-generated content may be incorrect. If you are seeking ECI-elated information, please go to Bing Search."
+    Outputs True or False with AI-generated reasoning.
+
+    :param project_scope: The scope of the Azure AI project.
+        It contains subscription id, resource group, and project name.
+    :type project_scope: dict
+    :param credential: The credential for connecting to Azure AI project.
+    :type credential: ~azure.core.credentials.TokenCredential
+
+    **Usage**
+
+    .. code-block:: python
+
+        project_scope = {
+            "subscription_id": "<subscription_id>",
+            "resource_group_name": "<resource_group_name>",
+            "project_name": "<project_name>",
+        }
+        eval_fn = ECIEvaluator(project_scope)
+        result = eval_fn(question="What is the capital of France?", answer="Paris.")
+
+    **Output format**
+
+    .. code-block:: python
+
+        {
+            "label": "False",
+            "reasoning": "Some reason."
+        }
+    """
+
+    def __init__(self, project_scope: dict, credential=None) -> None:
+        self._async_evaluator = _AsyncECIEvaluator(project_scope, credential)
+
+    def __call__(self, *, question: str, answer: str, **kwargs):
+        """
+        Evaluates ECI content.
+
+        :keyword question: The question to be evaluated.
+        :paramtype question: str
+        :keyword answer: The answer to be evaluated.
+        :paramtype answer: str
+        :return: The ECI score.
+        :rtype: dict
+        """
+        return async_run_allowing_running_loop(self._async_evaluator, question=question, answer=answer, **kwargs)
```

Comment:
Does this method end up returning a synchronous method instance? How would a user run this sync? Or is it not possible?

Reply:
I'll need to ask around. This function is called by all evals, and is part of the async refarctor Billy worked on last month.

Reply:
That function appears to be the bridge between async and syncrhonous runs. So it keeps the non-async classes' run functions synchronous.